brave/brave-core
# Check result from `ProfileSyncServiceFactory::GetForProfile`

which will be null when --disable-sync specified

Also remove redundant `--disable-brave-sync` switch because brave sync relies
on chromium sync, it won't work properly with chromium sync disabled.

fix https://github.com/brave/brave-browser/issues/8108

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. Launch Brave with `--disable-sync` 
2. Brave won't crash at startup
3. Check sync is completely disabled,
3.1. User has to access to sync on menu
3.2 `brave::sync` becomes inaccessible
3.3 `brave://inspect/#extensions` has no Brave Sync extension running
4. Launch Brave with `--disable-brave-sync` 
5. Sync should be enabled because the switch is no longer valid

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
